zip-0227.rst

```diff
@@ -9,7 +9,614 @@
++------------------------------------+--------------------------+-------------------------------------------+---------------------------------------------------------------------------+
+|``64``                              |``issueAuthSig``          |``byte[64]``                               |The signature of the transaction SIGHASH, signed by the issuer,            |
+|                                    |                          |                                           |validated as in `Issuance Authorization Signature Scheme`_.                |
++------------------------------------+--------------------------+-------------------------------------------+---------------------------------------------------------------------------+
+
+Issuance Protocol
+-----------------
+The issuer program performs the following operations:
+
+For all actions ``IssueAction``:
+
+- encode :math:`\mathsf{asset\_desc}` as a UTF-8 byte string of size up to 512.
+- compute :math:`\mathsf{AssetDigest}` from the issuance validating key :math:`\mathsf{ik}` and :math:`\mathsf{asset\_desc}` as decribed in the `Specification: Asset Identifier`_ section.
+- compute :math:`\mathsf{AssetBase}` from :math:`\mathsf{AssetDigest}` as decribed in the `Specification: Asset Identifier`_ section.
+- set the :math:`\mathsf{finalize}` boolean as desired (if more issuance actions are to be created for this :math:`\mathsf{AssetBase}`, set :math:`\mathsf{finalize} = 0`, otherwise set :math:`\mathsf{finalize} = 1`).
+- for each recipient :math:`i`:
+
+    - generate a ZSA output note that includes the Asset Base. For an Orchard-ZSA note this is :math:`\mathsf{note}_i = (\mathsf{d}_i, \mathsf{pk}_{\mathsf{d}_i}, \mathsf{v}_i, \rho_i, \mathsf{rseed}_i, \mathsf{AssetBase}, \mathsf{rcm}_i)\!`.
+
+- encode the ``IssueAction`` into the vector ``vIssueActions`` of the bundle.
+
+For the ``IssueBundle``:
+
+- encode the ``vIssueActions`` vector.
+- encode the :math:`\mathsf{ik}` as 32 byte-string.
+- sign the SIGHASH transaction hash with the issuance authorizing key, :math:`\mathsf{isk}`, using the :math:`\mathsf{IssueAuthSig}` signature scheme. The signature is then added to the issuance bundle.
+
+
+**Note:** that the commitment is not included in the ``IssuanceAction`` itself. As explained below, it is computed later by the validators and added to the note commitment tree.
+
+
+Specification: Consensus Rule Changes
+=====================================
+
+For the ``IssueBundle``:
+
+- Validate the issuance authorization signature, :math:`\mathsf{issueAuthSig}`, on the SIGHASH transaction hash, :math:`\mathsf{SigHash}`, by invoking :math:`\mathsf{IssueAuthSig.Validate(ik, SigHash, issueAuthSig)}`
+
+For each ``IssueAction`` in ``IssueBundle``:
+
+- check that :math:`0 < \mathtt{assetDescSize} <= 512`.
+- check that :math:`\mathsf{asset\_desc}` is a string of length :math:`\mathtt{assetDescSize}` bytes.
```

Comment:
It could not fail to be if the IssueAction parses correctly. (All parsing is an *implicit* consensus rule.)

Reply:
Removed this check, thanks